test/mox_test.exs

```diff
@@ -351,15 +351,57 @@ defmodule MoxTest do
 
       verify!()
       expect(CalcMock, :add, fn x, y -> x + y end)
+      expect(SciCalcOnlyMock, :exponent, fn x, y -> x * y end)
 
-      message = ~r"expected CalcMock.add/2 to be invoked once but it was invoked 0 times"
-      assert_raise Mox.VerificationError, message, &verify!/0
+      assert_raise(Mox.VerificationError, &verify!/0)
+      try do
+        verify!()
+      rescue
+        error in Mox.VerificationError ->
+          assert error.message =~ ~r"expected CalcMock.add/2 to be invoked once but it was invoked 0 times"
+          assert error.message =~ ~r"expected SciCalcOnlyMock.exponent/2 to be invoked once but it was invoked 0 times"
+      end
 
-      assert CalcMock.add(2, 3) == 5
+      CalcMock.add(2, 3)
+      assert_raise(Mox.VerificationError, &verify!/0)
+      try do
+        verify!()
+      rescue
+        error in Mox.VerificationError ->
+          refute error.message =~ ~r"expected CalcMock.add/2"
+          assert error.message =~ ~r"expected SciCalcOnlyMock.exponent/2 to be invoked once but it was invoked 0 times"
+      end
+
+      SciCalcOnlyMock.exponent(2, 3)
       verify!()
+
+      # Adding another expected call makes verification fail again
       expect(CalcMock, :add, fn x, y -> x + y end)
+      assert_raise(Mox.VerificationError, &verify!/0)
+      try do
+        verify!()
+      rescue
+        error in Mox.VerificationError ->
+          assert error.message =~ ~r"expected CalcMock.add/2 to be invoked 2 times but it was invoked once"
+          refute error.message =~ ~r"expected SciCalcOnlyMock.exponent/2"
+      end
+    end
 
-      message = ~r"expected CalcMock.add/2 to be invoked 2 times but it was invoked once"
+    test "verifies when mocks are over-called in the process in private mode" do
```

Comment:
**Annotation:** The `"verifies when mocks are over-called` tests are the tests that are currently failing & that would be useful to me in some of the code I'm working on.